test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -23,4 +23,88 @@ describe('Loader hooks', () => {
     assert.match(lines[2], /{"url":"file:\/\/\/.*\/experimental\.json","format":"test","shortCircuit":true}/);
     assert.match(lines[3], /{"source":{"type":"Buffer","data":\[.*\]},"format":"json","shortCircuit":true}/);
   });
+
+  it('should be able to handle never resolving dynamic imports (ESM entry point)', async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--experimental-loader',
+      fixtures.fileURL('es-module-loaders/never-settling-resolve-step/loader.mjs'),
```

Comment:
Not for this PR, but I've noticed that in most uses of `fixtures.fileURL` we treat it as if it's returning a string when in fact it's a URL, and I guess it's getting cast to a string here. We might want to change the function.

Reply:
Does it matter? FWIW I prefer it returning a `URL` as now.